src/routes/OAuth/Login/Login.js

```diff
@@ -0,0 +1,98 @@
+import { push } from 'connected-react-router'
+import React, { useEffect, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { Link, useParams } from 'react-router-dom'
+import { formatError } from 'routes/NonAuthLayoutRouter/util'
+import TextInput from 'components/TextInput'
+import Button from 'components/Button'
+import { login } from './Login.store'
+import './Login.scss'
+
+export default function Login (props) {
+  const dispatch = useDispatch()
+
+  const params = useParams()
+
+  const oauthUID = params.uid
+
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState(null)
+
+  if (props.authenticated) {
+    useEffect(() => {
+      // If already authenticated then do the oAuth login with current session user
+      submit()
+    }, [])
+  }
+
+  const submit = async () => {
+    try {
+      const { payload } = await dispatch(login(oauthUID, email, password))
+      const { redirectTo } = payload
+      if (redirectTo) {
+        window.location.href = redirectTo
+      } else {
+        dispatch(push('/'))
+      }
+    } catch (e) {
+      setError(e.message)
+    }
+  }
+
+  // TODO: let people log in with Google to login with Hylo? 🤪
```

Comment:
if we do... might just be easier to do this as a routing redirect, no? Assuming that once they are logged in, they can just follow the logic denoted on line 24.

Reply:
yes if they are already logged in they just go through the consent flow. now that i think about it we probably do need log in with google here in case they originally signed up to Hylo with Google. it just feels really weird to have someone go through the Google oAuth flow only to then go through ours... 🤔

Reply:
i think ill leave it this way unless/until it becomes an issue